opal/mca/accelerator/rocm/accelerator_rocm_component.c

```diff
@@ -31,7 +36,12 @@ size_t opal_accelerator_rocm_memcpyH2D_limit=1048576;
 static opal_mutex_t accelerator_rocm_init_lock;
 static bool accelerator_rocm_init_complete = false;
 
-hipStream_t opal_accelerator_rocm_MemcpyStream = NULL;
+/* Define global variables, used in accelerator_rocm.c */
+hipStream_t opal_accelerator_rocm_alloc_stream = NULL;
+opal_mutex_t opal_accelerator_rocm_stream_lock = {0};
+int opal_accelerator_rocm_num_devices = 0;
+float *opal_accelerator_rocm_mem_bw = NULL;
+hipStream_t *opal_accelerator_rocm_MemcpyStream = NULL;
```

Comment:
actually, is this change from `hipStream_t opal_accelerator_rocm_MemcpyStream` to `hipStream_t *opal_accelerator_rocm_MemcpyStream` truly correct? We do not have anywhere the memory allocated for the actualy hipStream_t object

Reply:
I made the change because of this line: https://github.com/open-mpi/ompi/pull/12356/files#diff-a83aa406fc5f195f910c83df9369e84a3202b9b10476dac3659fae9b4345453eL217

We shouldn't cast an opaque type to `void*`, that opens the door for all kinds of nasty behavior.

The memory is allocated here: https://github.com/open-mpi/ompi/pull/12356/files#diff-a83aa406fc5f195f910c83df9369e84a3202b9b10476dac3659fae9b4345453eR198

Reply:
Ahh yes, I believe this was a relict from when I experimented with memory pools (which ultimately didn't seem to help). There should only be `opal_accelerator_rocm_MemcpyStream` left now.